### ISSUE TITLE
Vector2's floor() causes exported executable to close immediately

### ISSUE THREAD

**Issue by ejnij** [godotengine/godot#21122]
**Godot version:**

3.1.dev.calinou.68676c7 (downloaded it today)
Update: still occurs with  v3.1.alpha.calinou.9eb4d4a
**OS/device including version:**

Windows 10 64-bit

**Issue description:**

I'm gonna tell the whole story, so I can share my saltiness with you. To save your precious time there's a TL;DR at the end.
So it started like this - I had a problem that was fixed by using a dev build (Thanks for helping me with that btw). Unfortunately, I wasn't able to export an executable that worked - it immediately closed after the Godot splash screen. I then tried making a new project, and then was able to export properly. So what did naive me do? I thought the problem was some kind of incompatibility with the project files, and decided to re-do everything in a new project - obviously I copied all the scripts, but it still took much more than I wanted it to and was quite troublesome. Finally, I was done! I was sure everything would work this way, but no! The problem was still there. The next logical step was to see which Scene was the problem, so I went over every scene, set it as main, exported it, and checked if the problem occurs.
After making sure there's no mistake, I found that the Player's scene was the problem. I started removing nodes & scenes inside the Player's scene till I had none, but it still didn't work! So I removed the script. It worked! Okay, so the problem is the script, but which part? I started removing methods until I ended up removing the problematic one - it was a simple method that did a basic math calculation for diagonal jump vector (for wall-jumps). I was really surprised, is this really what was causing me all these problems? So I start breaking it apart part by part, eventually figuring out it was the floor() at the end. I shed a tear. The end.

TL;DR:
Basically it's as the title states - calling floor() on a Vector2 type causes the executable to close immediately. I've added a super basic example in the project below.

I would like to add that I did not have this problem in 3.0.5.

Update:
It might be a combination of multiple Vector methods, as unlike on the previous version, this time it only occurs with a line of code that's similar to the one I have in my project, which also includes rotated().
Another finding is that it doesn't happen in a debug executable. (which wasn't the case earlier)

**Steps to reproduce:**
1) Use the floor method on a Vector2 type.
2) Cry when your exported executable closes immediately.

**Minimal reproduction project:**

[~~floor_ruined_my_life.zip~~](https://github.com/godotengine/godot/files/2298040/floor_ruined_my_life.zip)
Update: New version -
[Floor Still Ruins My Life.zip](https://github.com/godotengine/godot/files/2339611/Floor.Still.Ruins.My.Life.zip)

**Comment by Chaosus** [comment]
_Checked with both release and debug export templates, in current master its fixed. Closed by bugsquad_

**Comment by akien-mga** [comment]
Reopening, one tester being unable to reproduce it doesn't mean that it's fixed, especially if it was reported only 12 days ago.

**Comment by akien-mga** [comment]
That said, I can't reproduce the issue either on Linux x86_64.

Does the issue happen when running the project from the editor, or only once exported? If the latter, are you using export templates that match the commit you're using for the editor?

Please post the output from running the project in a terminal.

**Comment by ejnij** [comment]
It's the latter, and yea I did. Is there anywhere I can download that version again? I've since removed it. I can also confirm it doesn't happen with the latest nightly build.

**Comment by ejnij** [comment]
I updated my new findings and the reproduction project.
TL;DR - Still occurs with yesterday's nightly build, but not in debug executable. Also, this time just 'floor()' isn't enough, so I just copied the actual line that causes the bug in my project. (some variables changed to hard-coded numbers)

**Comment by akien-mga** [comment]
Thanks, I could reproduce the bug with 3.1-alpha1 released today, exporting a release binary for Linux x86_64 with the provided project.

Stacktrace:
```
(gdb) bt
#0  0x00000000017af909 in Variant::get_method_return_type(Variant::Type, StringName const&, bool*) ()
#1  0x00000000008cbc3b in GDScriptParser::_reduce_function_call_type(GDScriptParser::OperatorNode const*) ()
#2  0x00000000008cd193 in GDScriptParser::_reduce_node_type(GDScriptParser::Node*) ()
#3  0x00000000008d436a in GDScriptParser::_check_block_types(GDScriptParser::BlockNode*) ()
#4  0x0000000000000000 in ?? ()
```

CC @vnen

**Comment by vnen** [comment]
Hm, this _should_ be available in release (at least in my opinion).

**Comment by vnen** [comment]
Can't reproduce this with a recent `master`. Can you try with [beta 1](https://godotengine.org/article/dev-snapshot-godot-3-1-beta-1)? Couldn't reproduce with that either.

**Comment by akien-mga** [comment]
Assuming fixed by recent GDScript fixes.

Please comment if you can still reproduce it in 3.1 beta 2 or later.
